### PR TITLE
Add QPainterCompositionMode enum + method

### PR DESCRIPTION
**Pull request by Montel**

### CODE REVIEW

File: crates/cxx-qt-lib/src/gui/qpainter.rs [hunk 1]
```diff
@@ -189,6 +289,10 @@ mod ffi {
         #[rust_name = "set_clip_rect"]
         fn setClipRect(self: Pin<&mut QPainter>, rectangle: &QRect, operation: ClipOperation);
 
+        /// Sets the composition mode to the given mode.
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-lib/src/gui/qpainter.rs:292]: wonder if this warning should be somewhere?

```
Warning: Only a [QPainter](https://doc.qt.io/qt-6/qpainter.html) operating on a [QImage](https://doc.qt.io/qt-6/qimage.html) fully supports all composition modes. The RasterOp modes are supported for X11 as described in [compositionMode](https://doc.qt.io/qt-6/qpainter.html#compositionMode)().
```

and the https://doc.qt.io/qt-6/qpainter.html#CompositionMode-enum has a lot more description too

File: crates/cxx-qt-lib-headers/include/gui/qpainter.h [hunk 2]
```diff
@@ -11,8 +11,13 @@
 
 #include <QtGui/QPainter>
 
+#include "rust/cxx.h"
+
+#include <cstdint>
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-lib-headers/include/gui/qpainter.h:16]: why is cstdint and rust/cxx.h required here? I thought maybe for the shared enum but `qt.h` doesn't have these and those work ?

Comment (Montel) [reply crates/cxx-qt-lib-headers/include/gui/qpainter.h:16]: I think it's a copy/paste error.

**Comment by ahayzen-kdab** [comment]
@Montel rebase this one after the other one lands as it needs manual fixing.

**Pull request merged**